src/components/elements/Map-mapbox/Map.tsx

```diff
@@ -57,114 +94,208 @@ export const Map = ({
   hasControls = true,
   additionalPolygonProperties,
   captureAdditionalPolygonProperties,
+  siteData = false,
+  status = false,
+  editPolygon = false,
+  polygonChecks = false,
+  record,
+  isUserDrawing = false,
+  setIsUserDrawing,
+  showPopups = false,
+  showLegend = false,
+  mapFunctions,
+  tooltipType = "view",
   ...props
 }: MapProps) => {
-  const onError = useDebounce((hasError, errors) => _onError?.(hasError, errors), 250);
+  const [viewImages, setViewImages] = useState(false);
+  const [currentStyle, setCurrentStyle] = useState(MapStyle.Satellite);
+  const { polygonsData, bbox, setPolygonFromMap, polygonFromMap, sitePolygonData } = props;
+  const context = useSitePolygonData();
+  const { isUserDrawingEnabled } = isUserDrawing
+    ? { isUserDrawingEnabled: isUserDrawing }
+    : context || { isUserDrawingEnabled: false };
+  const { toggleUserDrawing, toggleAttribute, reloadSiteData } = context || {};
+  if (!mapFunctions) {
+    return null;
+  }
+  const { map, mapContainer, draw, onCancel, styleLoaded, initMap, setStyleLoaded, setChangeStyle, changeStyle } =
+    mapFunctions;
 
-  const validateGeoJSON = function (map: IMap, source: string) {
-    if (!editable) return;
+  useEffect(() => {
+    initMap();
+  });
```

Comment:
Is this meant to run on every render? If it's supposed to be once on mount only, `useEffect` requires a second argument of `[]` in the dependencies param.

Reply:
Done.